components/movementsensor/rtkutils/ntrip.go

```diff
@@ -60,3 +92,113 @@ func NewNtripInfo(cfg *NtripConfig, logger logging.Logger) (*NtripInfo, error) {
 	logger.Debug("Returning n")
 	return n, nil
 }
+
+// ParseSourcetable gets the sourcetable and parses it.
+func (n *NtripInfo) ParseSourcetable() (*Sourcetable, error) {
+	reader, err := n.Client.GetSourcetable()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := reader.Close(); err != nil {
+			log.Println("Error closing reader:", err)
+		}
+	}()
+
+	st := &Sourcetable{}
+	st.Streams = make([]Stream, 0, 200)
+	scanner := bufio.NewScanner(reader)
+	ln := ""
+
+Loop:
+	for scanner.Scan() {
+		ln := scanner.Text()
+		if strings.HasPrefix(ln, "#") || strings.HasPrefix(ln, "*") { // comment
+			continue
+		}
+		fields := strings.Split(ln, ";")
+		switch fields[0] {
+		case "CAS":
+			continue
+		case "NET":
+			continue
+		case "STR":
+			if str, err := ParseStream(ln); err == nil {
+				st.Streams = append(st.Streams, str)
+			}
+		case "ENDSOURCETABLE":
+			break Loop
+		default:
+			return nil, fmt.Errorf("%s: illegal sourcetable line: '%s'", n.URL, ln)
+		}
+	}
+
+	if ln != "ENDSOURCETABLE" {
+		return nil, fmt.Errorf("invalid sourcetable: missing string \"ENDSOURCETABLE\" in %s", ln)
+	}
+
+	return st, nil
+}
+
+// ParseStream parses a line from the sourcetable.
+func ParseStream(line string) (Stream, error) {
+	fields := strings.Split(line, ";")
+
+	if len(fields) < 19 {
+		return Stream{}, fmt.Errorf("missing fields at stream line: %s", line)
+	}
+
+	carrier, err := strconv.Atoi(fields[5])
+	if err != nil {
+		return Stream{}, fmt.Errorf("parsing the streams carrier in line: %s", line)
```

Comment:
```suggestion
		return Stream{}, fmt.Errorf("cannot parse the streams carrier in line: %s", line)
```

same for other errors.